JacquardToolkit/JacquardToolkit/JacquardService.swift

```diff
@@ -209,33 +217,38 @@ extension JacquardService: CBPeripheralDelegate {
             input_data![JSConstants.JSNumbers.ForceTouch.fullThreadCount - JSConstants.JSNumbers.ForceTouch.threadCount + i] = threadReadings[i] as NSNumber
         }
         
-        let prediction = try? forceTouchModel.prediction(input: NewGestureClassifier_RC2Input(_15ThreadConductivityReadings: input_data!))
-        print("Prediction Result: \((prediction?.output["ForceTouch"])!)")
-        
+        let prediction = try? forceTouchModel.prediction(input: ForcetouchInput(_15ThreadConductivityReadings: input_data!))
+//        print("Prediction Result: \((prediction?.output["ForceTouch"])!)")
+
         if forceTouchTurnedEnabled {
-            if ((prediction?.output["ForceTouch"])! > 0.7) {
-                forceTouchTurnedEnabled = false
-                delegate?.didDetectForceTouchGesture()
-                return true
+            if ((prediction?.output["ForceTouch"])! > forceTouchDetectionThreshold) {
+                forceTouchDetectionProgress += 1
+                
+                if forceTouchDetectionProgress >= forceTouchDetectionLength {
+                    forceTouchDetectionProgress = 0
+                    forceTouchTurnedEnabled = false
+                    delegate?.didDetectForceTouchGesture!()
+                    return true
+                }
+                
+            } else {
+                forceTouchDetectionProgress = 0
             }
         } else {
-            //add the next confidence interval into the array
-            confidenceIntervalArray.append(prediction?.output["ForceTouch"] ?? 0.0)
-            if confidenceIntervalArray.count > 5 {
-                var reenableForceTouch = true
-                for i in confidenceIntervalArray {
```

Comment:
Better var name than i would be awesome. Maybe just confidenceInterval...

Reply:
this is your code that i deleted (it's replaced with a simplified approach?)

Reply:
You're right. I'm bad